### PR TITLE
Initial fx_renderer implementation

### PR DESCRIPTION
**Pull request by ErikReider**
Closes: #3 

For some reason the rendered surfaces are shaded yellow, but only when running with gdb? Maybe just on my system?

### CODE REVIEW

**Comment by WillPower3309** [comment]
In the "initial implementation" commit, did you make any changes to fx renderer or was it just a copy paste? Wondering if it's worth me looking over the diff for the fx renderer files from that commit to review anything there

**Comment by ErikReider** [comment]
> In the "initial implementation" commit, did you make any changes to fx renderer or was it just a copy paste? Wondering if it's worth me looking over the diff for the fx renderer files from that commit to review anything there

Some very minor changes if I remember correctly. So a simple diff would reveal the minor changes :)

**Comment by WillPower3309** [comment]
Oops sorry, misclick

**Comment by WillPower3309** [comment]
Fixed compilation errors, but I think I broke rendering along the way :S will take a look later

**Comment by WillPower3309** [comment]
Everything should be good to go!

**Comment by WillPower3309** [comment]
Pending a look over by @ErikReider , lets get this in, get #8 in, and then we can focus on adding in the other effects

**Review by WillPower3309 — APPROVED** [review]

File: include/render/fx_renderer/fx_renderer.h [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+enum fx_tex_shader_source {
+	SHADER_SOURCE_TEXTURE_RGBA = 1,
+	SHADER_SOURCE_TEXTURE_RGBX = 2,
+	SHADER_SOURCE_TEXTURE_EXTERNAL = 3,
+};
+
+struct decoration_data {
+	float alpha;
+	int corner_radius;
+};
```
Comment (WillPower3309) [inline include/render/fx_renderer/fx_renderer.h]: We should remove this before merging (need to rethink how we handle decoration data)

File: include/render/fx_renderer/fx_renderer.h [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+#define _FX_OPENGL_H
+
+#include <GLES2/gl2.h>
+#include <GLES2/gl2ext.h>
+#include <stdbool.h>
+#include <wlr/render/egl.h>
+#include <wlr/render/wlr_texture.h>
+#include <wlr/util/addon.h>
+
+#include "wlr/util/box.h"
```
Comment (WillPower3309) [inline include/render/fx_renderer/fx_renderer.h]: Should this be <wlr/util/box.h>?

**Pull request merged**